services/httppost/service.go

```diff
@@ -0,0 +1,253 @@
+			e.SetHeaders(c.Headers)
+
+			endpointSet[c.Endpoint] = true
+		} else {
+			return fmt.Errorf("unexpected config object type, got %T exp %T", nc, c)
+		}
+	}
+
+	// Find any deleted endpoints
+	for name, endpoint := range s.endpoints {
+		if !endpointSet[name] {
+			endpoint.Close()
+			delete(s.endpoints, name)
+		}
+	}
+
+	return nil
+}
+
+func (s *Service) Open() error {
+	return nil
+}
+
+func (s *Service) Close() error {
+	return nil
+}
+
+type testOptions struct {
+	Endpoint string            `json:"endpoint"`
+	URL      string            `json:"url"`
+	Headers  map[string]string `json:"headers"`
+}
+
+func (s *Service) TestOptions() interface{} {
+	return &testOptions{
+		Endpoint: "example",
+		URL:      "http://localhost:3000/",
+		Headers:  map[string]string{"Auth": "secret"},
+	}
+}
+
+func (s *Service) Test(options interface{}) error {
+	var err error
+	o, ok := options.(*testOptions)
+	if !ok {
+		return fmt.Errorf("unexpected options type %t", options)
+	}
+
+	event := alert.Event{}
+	body := bytes.NewBuffer(nil)
+	ad := alertDataFromEvent(event)
+
+	err = json.NewEncoder(body).Encode(ad)
+	if err != nil {
+		return fmt.Errorf("failed to marshal alert data json: %v", err)
+	}
+
+	// Create the HTTP request
+	var req *http.Request
+	e := &Endpoint{
+		url:     o.URL,
+		headers: o.Headers,
+	}
+	req, err = e.NewHTTPRequest(body)
+
+	// Execute the request
+	req.Header.Set("Content-Type", "application/json")
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to POST alert data: %v", err)
+	}
+	resp.Body.Close()
+	return nil
+}
+
+type HandlerConfig struct {
+	URL      string `mapstructure:"url"`
+	Endpoint string `mapstructure:"endpoint"`
+}
+
+type handler struct {
+	s        *Service
+	bp       *bufpool.Pool
+	endpoint *Endpoint
+	logger   *log.Logger
+}
+
+func (s *Service) Handler(c HandlerConfig, l *log.Logger) alert.Handler {
+	e, ok := s.Endpoint(c.Endpoint)
+	if !ok {
+		e = NewEndpoint(c.URL, nil)
+	}
+
+	return &handler{
+		s:        s,
+		bp:       bufpool.New(),
+		endpoint: e,
+		logger:   l,
+	}
+}
+
+func (h *handler) NewHTTPRequest(body io.Reader) (req *http.Request, err error) {
+	return h.endpoint.NewHTTPRequest(body)
+}
+
+func (h *handler) Handle(event alert.Event) {
+	var err error
+
+	// Construct the body of the HTTP request
+	body := h.bp.Get()
+	defer h.bp.Put(body)
+	ad := alertDataFromEvent(event)
+
+	err = json.NewEncoder(body).Encode(ad)
+	if err != nil {
+		h.logger.Printf("E! failed to marshal alert data json: %v", err)
+		return
+	}
+
+	req, err := h.NewHTTPRequest(body)
+	if err != nil {
+		h.logger.Printf("E! fail to create HTTP request: %v", err)
+		return
+	}
+
+	// Execute the request
+	req.Header.Set("Content-Type", "application/json")
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		h.logger.Printf("E! failed to POST alert data: %v", err)
+		return
+	}
+	resp.Body.Close()
+}
+
+func alertDataFromEvent(event alert.Event) alert.Data {
```

Comment:
Great, you can move it to the alert package now too. Maybe even just make it a method of the Event type, but what ever looks right to you.